Sentinel/Classes/UserDefaults/UserDefaultsToolViewModel.swift

```diff
@@ -11,26 +11,70 @@ final class UserDefaultsToolViewModel: ObservableObject {
 
     // MARK: - Internal properties
 
-    let value: String
-    let title: String
-    let userDefaults: UserDefaults
+    @Published var sections: [ToolTableSection] = []
+    let name: String
+
+    // MARK: - Private properties
+
+    private let userDefaults: UserDefaults
 
     // MARK: - Init
 
-    init(value: String, title: String, userDefaults: UserDefaults = .standard) {
-        self.value = value
-        self.title = title
+    init(name: String, userDefaults: UserDefaults = .standard) {
+        self.name = name
         self.userDefaults = userDefaults
     }
 
+    func updateSections() {
+        sections = createSectionItems(with: userDefaults)
+    }
 }
 
-// MARK: - Internal methods
+// MARK: - Item creation
 
-extension UserDefaultsToolViewModel {
+private extension UserDefaultsToolViewModel {
 
-    func didPressDelete() {
-        UserDefaults.standard.removeObject(forKey: title)
-    }
+    func createSectionItems(with userDefaults: UserDefaults) -> [ToolTableSection] {
+        let items = userDefaults.dictionaryRepresentation()
+            .sorted { $0.key < $1.key }
+            .map { (key, value) in
+                #if os(macOS)
+                ToolTableItem.navigation(
+                    NavigationToolItem(
+                        title: key,
+                        didSelect: {
+                            UserDefaultsToolDetailView(
+                                viewModel: UserDefaultsToolDetailViewModel(
+                                    value: String(describing: value),
+                                    title: key,
+                                    userDefaults: userDefaults,
+                                    didDeleteProperty: { [unowned self] in sections = createSectionItems(with: userDefaults) }
+                                ),
+                                selection: $0
+                            )
+                        }
+                    )
+                )
+                #else
+                ToolTableItem.navigation(
+                    NavigationToolItem(
+                        title: key,
+                        didSelect: {
+                            UserDefaultsToolDetailView(
+                                viewModel: UserDefaultsToolDetailViewModel(
+                                    value: String(describing: value),
+                                    title: key,
+                                    userDefaults: userDefaults,
+                                    didDeleteProperty: nil // onAppear will update the screen on iOS
+                                )
+                            )
+                        }
+                    )
+                )
+                #endif
```

Comment:
I would prefer moving `didSelect` into separate method, or just defining `didDeleteProperty` as a standalone variable and init the variable based on `if-else`.

Reply:
The difference between didSelect closures is messing up this one. The macOS one has a value provided in it which is then sent as the selection parameter, and the iOS one doesn't. Nothing comes to mind on how to handle this elegantly. Maybe you might think of something